lib/proxy_api/external_ipam.rb

```diff
@@ -83,101 +81,96 @@ def next_ip(subnet, mac, group = "")
     #
     # Returns: true if IP was added successfully to External IPAM, otherwise false
     #
-    # Responses from Proxy plugin:
+    # Responses:"
     #   Response if success:
-    #     Net::HTTPCreated
+    #     201
     #   Response if IP already reserved:
     #     {"error": "IP address already exists"}
     #   Response if subnet error:
-    #     {"error": "The specified subnet does not exist in External IPAM."}
+    #     {"error": "Subnet not found in External IPAM"}
     #   Response if missing required params:
     #     {"error": ["A 'cidr' parameter for the subnet must be provided(e.g. 100.10.10.0/24)","Missing 'ip' parameter. An IPv4 address must be provided(e.g. 100.10.10.22)"]}
     #   Response if can't connect to External IPAM server
     #     {"error": "Unable to connect to External IPAM server"}
     def add_ip_to_subnet(ip, subnet, group = "")
       raise "subnet cannot be nil" if subnet.nil?
+      raise "ip cannot be nil" if ip.nil?
       response = parse post({}, "/subnet/#{subnet}/#{ip}?group=#{URI.escape(group.to_s)}")
-      raise(response['error']) if response['error'].present?
-      response
+      process(response)
     rescue => e
       raise ProxyException.new(url, e, N_("Unable to add IP %{ip} to the subnet %{subnet} in External IPAM."), ip: ip, subnet: subnet)
     end
 
-    # Get a list of groups from External IPAM. A group is analagous to a 'section' in phpIPAM, and
-    # is a logical grouping of subnets/ips.
+    # Get a list of groups from External IPAM. A group is a logical grouping of subnets/ips.
     #
-    # Params: None
+    # Params:  None
     #
     # Returns: An array of groups on success, or a hash with a "error" key
     #          containing error on failure.
     #
-    # Responses from Proxy plugin:
+    # Responses:
     #   Response if success:
-    #     {"data": [
+    #     {[
     #       {name":"Test Group","description": "A Test Group"},
     #       {name":"Awesome Group","description": "A totally awesome Group"}
     #     ]}
     #   Response if no groups exist:
-    #     {"data": []}
+    #     {"error": "Group not found in External IPAM"}
```

Comment:
Returning empty elements like arrays or hashes is also a good practice I would not be getting rid of. Exactly vice versa!